addons/mrp/report/mrp_report_mo_overview.py

```diff
@@ -82,7 +82,14 @@ def _get_report_data(self, production_id):
 
         if production.bom_id:
             currency = (production.company_id or self.env.company).currency_id
-            missing_components = (bom_line for bom_line in production.bom_id.bom_line_ids if bom_line not in (production.move_raw_ids.bom_line_id + self._get_kit_bom_lines(production.bom_id)))
+            current_bom_lines = production.move_raw_ids.bom_line_id | self._get_kit_bom_lines(production.bom_id)
+            missing_components = production.bom_id.bom_line_ids.filtered(
+                lambda bom_line: bom_line not in current_bom_lines and
+                (not bom_line.bom_product_template_attribute_value_ids or
+                    any(variant in production.product_id.product_template_attribute_value_ids.ids
+                    for variant in bom_line.bom_product_template_attribute_value_ids.ids)
+                )
+            )
```

Comment:
Still got an issue with this, here are my product attributes:
![image](https://github.com/user-attachments/assets/e8428426-1902-4af9-90c8-5975095263be)
And here is its BoM:
![image](https://github.com/user-attachments/assets/6067e81f-6b01-4e17-940c-1176e99c2e82)

Now, when doing a MO for a Variable (Green, M), you should have only two components. Both the BoM Overview and the MO components are correct. Yet, when opening the overview, it will still count the third component:
![image](https://github.com/user-attachments/assets/f93ac3a1-1696-4f46-829b-1cc72f8c0467)